src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/modifier/modifierorder/InputModifierOrderSealedAndNonSealed.java

```diff
@@ -0,0 +1,57 @@
+//non-compiled with javac: Compilable with Java15
+package com.puppycrawl.tools.checkstyle.checks.modifier.modifierorder;
+
+/* Config:
+ * default
+ */
+
+sealed public class InputModifierOrderSealedAndNonSealed // violation
+    permits Circle, Square, Rectangle {
+}
+
+final class Circle extends InputModifierOrderSealedAndNonSealed implements Squircle {
+}
+
+sealed class Rectangle extends InputModifierOrderSealedAndNonSealed // ok
+    implements Cloneable
+    permits TransparentRectangle, FilledRectangle {
+}
+
+final class TransparentRectangle extends Rectangle {
+}
+
+sealed class Square extends InputModifierOrderSealedAndNonSealed implements Squircle {
+    sealed private class OtherSquare extends Square permits OtherSquare2 { // violation
+    }
+
+    private final class OtherSquare2 extends OtherSquare {
+    }
+}
+
+final strictfp class FilledRectangle extends Rectangle {
+}
+
+sealed interface Squircle permits Circle, Square { // ok
+}
+
+strictfp sealed interface Rhombus permits Parallelogram, Parallelogram.Diamond,
+        Parallelogram.Trapeziod {
+
+}
+
+record Parallelogram(int x, int y, double z) implements Rhombus{
+    final public record Diamond(int x, int y, double z)implements Rhombus { // violation
+    }
+
+    strictfp public final record Trapeziod(int x, int y, double z)implements Rhombus {
+    }
+}
+
+strictfp non-sealed interface Ellipse extends Square { // violation
```

Comment:
Please remove this as this is non-compilable.

Reply:
This needed to extend `Squircle`, fixed.

Reply:
Resolved